pkg/config/signer.go

```diff
@@ -82,32 +81,30 @@ func (s *Signer) Serialize() ([]byte, error) {
 	// serialize identities by prepending the MSPID and appending
 	// the ASN.1 DER content of the cert
 	sID := &msp.SerializedIdentity{Mspid: s.mspID, IdBytes: pemBytes}
-	idBytes := MarshalOrPanic(sID)
+	idBytes, err := proto.Marshal(sID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal serialized indentity: %v", err)
+	}
 
 	return idBytes, nil
 }
 
-// Public returns the public key of the signer.
-func (s *Signer) Public() crypto.PublicKey {
-	return s.publicKey
-}
-
 // Cert returns the certificate of the signer.
-func (s *Signer) Cert() *x509.Certificate {
+func (s *SigningIdentity) Cert() *x509.Certificate {
 	return s.cert
 }
 
 // MSPId returns the MSP ID of the signer.
-func (s *Signer) MSPId() string {
+func (s *SigningIdentity) MSPId() string {
 	return s.mspID
 }
 
-// Sign performs ECDSA sign with signer's private key on given digest. It
-// ensures signatures are created with Low S values since Fabric normalizes
+// Sign performs ECDSA sign with SigningIdentity's private key on given digest.
+// It ensures signatures are created with Low S values since Fabric normalizes
 // all signatures to Low S.
 // See https://github.com/bitcoin/bips/blob/master/bip-0146.mediawiki#low_s
 // for more detail.
-func (s *Signer) Sign(reader io.Reader, digest []byte) (signature []byte, err error) {
+func (s *SigningIdentity) Sign(reader io.Reader, digest []byte) (signature []byte, err error) {
 	if reader == nil {
 		return nil, errors.New("failed to sign, reader can not be nil")
```

Comment:
The consumer of this function knows that it called Sign and should handle any further error handling/reporting with that string. We can simplify this error to `errors.New("reader can not be nil")`, test will need to be updated as well

Reply:
done